### PR TITLE
[EXC-3285] Update Maven docs

### PR DESCRIPTION
**Pull request by joaquinstan**
- Remove mention of possibility of publishing APIs
- Update what types of assets can be published
- Update what types of assets can be consumed
- Add section that clarifies what types assets are shown in the Exchange UI 
- Add clarification for external dependencies

### CODE REVIEW

File: modules/ROOT/pages/to-publish-assets-maven.adoc [hunk 1]
```diff
@@ -4,7 +4,13 @@ include::_attributes.adoc[]
 endif::[]
 :imagesdir: ../assets/images/
 
-Anypoint Exchange provides the https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/exchange-maven-facade-api-http/[Maven Facade] that enables Maven clients to publish and consume assets using Maven. The Maven Facade can publish connectors, examples, and templates as Exchange assets. You can publish APIs directly in Exchange. You can also use the Maven Facade to publish APIs.
+Anypoint Exchange provides the https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/exchange-maven-facade-api-http/[Maven Facade] that enables Maven clients to publish and consume assets using Maven. Using the Maven Facade, you can publish the following as Exchange assets:
+
+* Connectors
```
Comment (LucasLudueno) [inline modules/ROOT/pages/to-publish-assets-maven.adoc]: and libraries? 🤔 do we want to expose it @juancoen?

File: modules/ROOT/pages/to-publish-assets-maven.adoc [hunk 2]
```diff
@@ -197,9 +203,18 @@ https://anypoint.mulesoft.com/accounts/api/profile
 [IMPORTANT]
 The value in `<username>` is a literal, don't change it. The value tells the platform that you're using a token. Set your token in `<password>ACCESS_TOKEN</password>`.
 
+
+== Handling dependencies external to Exchange
+
+For the time being, it is not possible to publish Java libraries and POM files as Exchange assets. Due to this, assets that declare dependencies that are not published in Exchange are not currently supported. You can avoid this issue for templates, examples and Mule applications by publishing a https://stackoverflow.com/a/16222971[fat jar], which already includes all of the project dependencies.
```
Comment (juancoen) [inline modules/ROOT/pages/to-publish-assets-maven.adoc]: have you tested that this works in tooling service?

**Review by wwarner-salesforce — APPROVED** [review]

File: modules/ROOT/pages/to-publish-assets-maven.adoc [hunk 3]
```diff
@@ -197,9 +203,20 @@ https://anypoint.mulesoft.com/accounts/api/profile
 [IMPORTANT]
 The value in `<username>` is a literal, don't change it. The value tells the platform that you're using a token. Set your token in `<password>ACCESS_TOKEN</password>`.
 
+
+== Dependencies external to Exchange
+
+Java libraries and POM files cannot currently be published as Exchange assets.
+
+Assets that declare dependencies that are not published in Exchange are not currently supported.
```
Comment (juancoen) [inline modules/ROOT/pages/to-publish-assets-maven.adoc]: This is not entirely true. It only applies to Mule 4 Extensions and only to "private" dependencies that don't exist in Maven Central or MuleSoft Maven repositories

Comment (wwarner-salesforce) [reply modules/ROOT/pages/to-publish-assets-maven.adoc]: @juancoen thank you! Is this correct?

"Java libraries and POM files cannot currently be published as Exchange assets.

A private dependency is a dependency that does not exist in Maven Central or the MuleSoft Maven repositories.

Mule 4 Extensions that declare private dependencies that are not published in Exchange are not currently supported."

**Review by mcampo — APPROVED** [review]
👍

**Review by juancoen — APPROVED** [review]

File: modules/ROOT/pages/to-publish-assets-maven.adoc (hunk 3 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/to-publish-assets-maven.adoc]: fixed by Joaquin

**Comment by kuntald** [comment]
1. Is there anything that we need to add regarding "unknown" asset types. Refer to the following [slack discussion](https://docs.google.com/document/d/1J3R3BrnoeVH6r_cYALUMqa6alkgnc4QbqMnpVJX5KBY/edit#heading=h.wqebdcqn30os). Has Maven anything to do with how an asset type can become "Unknown"? If so, can we clarify that in this part of the document?

2. Maybe this is for @wwarner-salesforce. Can we build a table that simplifies the visualization of what can/can't be uploaded via Maven, and what can/can't be viewable in Exchange? Kinda like [here](https://docs.google.com/document/d/1J3R3BrnoeVH6r_cYALUMqa6alkgnc4QbqMnpVJX5KBY/edit#heading=h.wqebdcqn30os).

3. Have we also addressed, or at least clarified the following problem statements identified by our Professional services team & multiple customer. All of these are mentioned in the [MBI](https://mulesoft-roadmap.aha.io/epics/EX-E-196)

### Key problems

- 1. [​Only Maven deployed artifacts can be downloaded by Maven​.](https://docs.google.com/document/d/19AOGlB7bHIeCNZZ_GxUaOMDi0GajdxUIiJT8fYncPxY/edit#heading=h.75zef5kpbtk5)
- 2. ​[A non-recognized asset(not (template or example)) uploaded via Maven, doesn't show the asset in Exchange UI. It's available via Graph API.​](https://docs.google.com/document/d/19AOGlB7bHIeCNZZ_GxUaOMDi0GajdxUIiJT8fYncPxY/edit#heading=h.uxczngkv1pg8)
- 3. No way to list, view or delete parent pom files, upon which a mule-extension depends. Associated JIRA tic.

**Comment by joaquinstan** [comment]
Hi @kuntald 

1. "unknown" is an asset type that Exchange uses for assets that are initialized but don't end up having any file associated with them. It doesn't have anything to do with how Maven works and the fact that customers are allowed to see them is a bug. Do you think we should include some reference to this bug to clarify?

2. @wwarner-salesforce 

3. i. I think a clarification should be made regarding this, I'll add it.
   ii. This was taken into account in the new `Document your Exchange Assets` section.
  iii. This was taken into account in `Dependencies external to Exchange`.

Maybe the sections addressing 3 ii and 3 iii could be improved? Do you think they are clear enough?

**Comment by wwarner-salesforce** [comment]
Yes, we can certainly add the table from the linked Google doc, or a modified version of it.

**Comment by kuntald** [comment]
I haven't seen the sections `Dependencies external to Exchange` and `Document your Exchange Assets`. Are they already live? If they already exist, but in different sections, outside of the Maven section, then maybe we can link to those sections from the Maven section? That way, everything & anything related to Maven is in one place. Just brainstorming, I'll let @wwarner-salesforce make the final call on this one.

**Comment by wwarner-salesforce** [comment]
Those sections were in this PR and have been live about an hour, on the Maven page:

https://docs.mulesoft.com/exchange/to-publish-assets-maven#dependencies-external-to-exchange
https://docs.mulesoft.com/exchange/to-publish-assets-maven#document-your-exchange-assets

I think that's a good spot for them, but I'm open to moving them or adding links if you think of a better option.

**Pull request merged**